src/csharp/Microsoft.Spark.Experimental/Sql/ExperimentalFunctions.cs

```diff
@@ -23,8 +23,8 @@ public static class ExperimentalFunctions
         /// A delegate that returns a <see cref="Column"/> for the result of the Vector UDF.
         /// </returns>
         public static Func<Column, Column> VectorUdf<T, TResult>(Func<T, TResult> udf)
-            where T : IArrowArray
-            where TResult : IArrowArray
+            where T : DataFrameColumn
```

Comment:
@pgovind Is is possible to introduce this as new APIs instead of changing the existing ones? We have few customers using this and it would be an easier transition for them. What do you think? Would it be too much work?

Reply:
I have this working on my machine now as new APIs. Cleaning it up now and putting it up for review in a bit